video/external/src/backend.rs

```diff
@@ -0,0 +1,241 @@
+use slotmap::SlotMap;
+use std::fs::File;
+use std::io::copy;
+use std::path::PathBuf;
+use swf::{VideoCodec, VideoDeblocking};
+
+enum ProxyOrStream {
+    /// These streams are passed through to the wrapped software
+    /// backend, accessed using the stored ("inner") handle,
+    /// which is completely internal to this backend.
+    Proxied(VideoStreamHandle),
+
+    /// These streams are handled by this backend directly.
+    Owned(VideoStream),
+}
+
+/// A video backend that falls back to the software backend for most codecs,
+/// except for H.264, for which it uses an external decoder.
+pub struct ExternalVideoBackend {
+    streams: SlotMap<VideoStreamHandle, ProxyOrStream>,
+    openh264_lib_filepath: Option<PathBuf>,
+    software: SoftwareVideoBackend,
+}
+
+impl Default for ExternalVideoBackend {
+    fn default() -> Self {
+        Self::new(None)
+    }
+}
+
+impl ExternalVideoBackend {
+    fn get_openh264_data() -> Result<(&'static str, &'static str), Box<dyn std::error::Error>> {
+        // Source: https://github.com/cisco/openh264/releases/tag/v2.4.1
+        match (std::env::consts::OS, std::env::consts::ARCH) {
+            ("linux", "x86") => Ok((
+                "libopenh264-2.4.1-linux32.7.so",
+                "dd0743066117d63e1b2abc56a86506e5",
+            )),
+            ("linux", "x86_64") => Ok((
+                "libopenh264-2.4.1-linux64.7.so",
+                "19c561386a9564f8510fcb7586b9d402",
+            )),
+            ("linux", "arm") => Ok((
+                "libopenh264-2.4.1-linux-arm.7.so",
+                "2274a1bbd13f32b7afe22092e44fa2b5",
+            )),
+            ("linux", "aarch64") => Ok((
+                "libopenh264-2.4.1-linux-arm64.7.so",
+                "2aa205f08077aa2d049032e0b56c5b84",
+            )),
+            ("macos", "x86_64") => Ok((
+                "libopenh264-2.4.1-mac-x64.dylib",
+                "9fefa1e0279a49b8a4e9cf6fc148bc0c",
+            )),
+            ("macos", "aarch64") => Ok((
+                "libopenh264-2.4.1-mac-arm64.dylib",
+                "41f59bb5696ffeadbfba3a8a95ec39b7",
+            )),
+            ("windows", "x86") => Ok((
+                "openh264-2.4.1-win32.dll",
+                "a9360e6dd1e24320c3d65a0c65bf14a4",
+            )),
+            ("windows", "x86_64") => Ok((
+                "openh264-2.4.1-win64.dll",
+                "c85406e6b73812ec3fb9da5f898c6a9e",
+            )),
+            (os, arch) => Err(format!("Unsupported OS/ARCH: {} {}", os, arch).into()),
+        }
+    }
+
+    pub fn get_openh264() -> Result<PathBuf, Box<dyn std::error::Error>> {
+        // See the license at: https://www.openh264.org/BINARY_LICENSE.txt
+        const URL_BASE: &str = "http://ciscobinary.openh264.org/";
+        const URL_SUFFIX: &str = ".bz2";
+
+        let (filename, md5sum) = Self::get_openh264_data()?;
+
+        let filepath = std::env::current_dir()?.join(filename);
```

Comment:
Let's not use current_dir please... that could be anywhere
Maybe our config storage place for now?
Or absolute worst case, the folder that the ruffle executable is in... but that's *not* likely to be the same as current_dir

Reply:
> Maybe our config storage place for now?

Ummh, I'd find it weird to put anything "executable" in there.

> Or absolute worst case, the folder that the ruffle executable is in...

That should be fine IMHO.

> but that's not likely to be the same as current_dir

Yeah, good point, I'm a bit too Linux-CLI-minded... :sweat_smile:

Reply:
Changed to:
```rust
        let filepath = std::env::current_exe()?
            .parent()
            .ok_or("Could not determine Ruffle location.")?
            .join(filename);
```

This way, we don't need to touch `.gitignore` either, as the lib will be put into `target`.